armi/nuclearDataIO/cccc/cccc.py

```diff
@@ -14,159 +14,30 @@
 
 """
 Standard interface files for reactor physics codes.
-
-This module is designed to read/write Fortran record-based binary files that
-comply with the format established by the Committee on Computer Code
-Coordination (CCCC). [CCCC-IV]_
-
-Notes
------
-A CCCC record consists of a leading and ending integer, which indicates the size of the record in
-bytes. As a result, it is possible to perform a check when reading in a record to determine if it
-was read correctly, by making sure the record size at the beginning and ending of a record are
-always equal.
-
-There are similarities between this code and that in the PyNE cccc subpackage.
-This is the original source of the code. TerraPower authorized the publication
-of some of the CCCC code to the PyNE project way back in the 2011 era. 
-This code has since been updated significantly to both read and 
-write the files.
-
-This was originally created following Prof. James Paul Holloway's alpha
-release of ccccutils written in c++ from 2001. 
 """
 import io
 import itertools
 import struct
 import os
 from copy import deepcopy
+from typing import List
 
 import numpy
 
 from armi.localization import exceptions
 from armi import runLog
 
+from .. import nuclearFileMetadata
+
 IMPLICIT_INT = "IJKLMN"
```

Comment:
hahaha, no way... does CCCC use the implicit type rules from fortran?

Reply:
Not sure if it does actually use them, but it does seem that way based on all the naming + types so far